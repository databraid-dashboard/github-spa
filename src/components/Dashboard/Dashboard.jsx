/* eslint-disable import/no-named-as-default, no-shadow, consistent-return */
import React from 'react';
import { Container, Grid, Header, Icon, Button } from 'semantic-ui-react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { renderRepos } from '../../actions/renderActions';
import Issues from '../Issues/Issues';
import PrTable from '../PrTable/PrTable';
import Milestones from '../Milestones/Milestones';
import './Dashboard.css';

export const Dashboard = ({ repoName, currentPage, renderRepos }) => {
  if (currentPage !== 'dashboard') {
    return <div />;
  }
  if (currentPage === 'dashboard') {
    return (
      <div>
        <Button icon onClick={() => renderRepos()}>
          <Icon name="arrow left" />
        </Button>
        <Container>
          <Grid padded>
            <Header as="h2" icon textAlign="center">
              <Icon name="github" />
              <Header.Content>
                {repoName}
              </Header.Content>
            </Header>
            <Grid.Row columns={3}>
              <Grid.Column>
                <PrTable />
              </Grid.Column>
              <Grid.Column>
                <Issues />
              </Grid.Column>
              <Grid.Column>
                <Milestones />
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Container>
      </div>
    );
  }
};

Dashboard.propTypes = {
  repoName: PropTypes.string,
  currentPage: PropTypes.string.isRequired,
  renderRepos: PropTypes.func.isRequired,
};

Dashboard.defaultProps = {
  repoName: null,
};

const mapStateToProps = state => ({
  repoName: state.currentPage.repoName,
  currentPage: state.currentPage.render,
});

export const mapDispatchToProps = dispatch => bindActionCreators({
  renderRepos,
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Dashboard);
