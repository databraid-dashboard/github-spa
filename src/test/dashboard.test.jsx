import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Dashboard } from '../components/Dashboard/Dashboard';
import rootReducer from '../reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Should render a Dashboard that holds Connected Components', () => {
  it.only('Should render the Dashboard', () => {
    const store = mockStore({ rootReducer });
    const renderRepos = jest.fn();
    const dashComponent = shallow(
      <Dashboard store={store} repoName={'Some Repo'} currentPage={'dashboard'} renderRepos={renderRepos} />,
    );

    // console.log('dashComponent', dashComponent);

    expect(shallowToJson(dashComponent)).toMatchSnapshot();
  });
  // it('Should have an <Issues/> component', ()=> {
  //   const store = mockStore({rootReducer});
  //   const dashComponent = shallow(
  //     <Dashboard store={store} />
  //   );
  //   console.log(mountedDashComponent);
  //   expect(shallowToJson(mountedDashComponent)).find('Grid.Column').childAt(0).to.equal('Issues')
  // })
  // it('Should have an <milestonesReducer/> component', ()=> {
  //
  //   expect(shallowToJson())
  // })
  // it('Should have an <PrTable/> component', ()=> {
  //
  //   expect(shallowToJson())
  // })
  // it('It should show a loading bar when unmounted', ()=> {
  //
  // })
});


//
// import AdminRewards from '../src/components/Admin/rewards/admin-rewards/rewards-container';
// import React from 'react';
// import ReactDOM from 'react-dom';
// import { shallow } from 'enzyme';
// import { shallowToJson } from 'enzyme-to-json';
// import thunk from 'redux-thunk';
// import configureMockStore from 'redux-mock-store';
// import lvlupApp from '../src/reducers/index';
// const middlewares = [thunk];
// const mockStore = configureMockStore(middlewares);
// test('AdminRewards should render', () => {
//   const store = mockStore({ lvlupApp });
//   const adminrewards = shallow(
//     <AdminRewards store={store} />,
//   );
//   expect(shallowToJson(adminrewards)).toMatchSnapshot();
// });
