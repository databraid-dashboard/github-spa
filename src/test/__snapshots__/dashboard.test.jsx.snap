// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a Dashboard that holds Connected Components Should render the Dashboard 1`] = `
<div>
  <Button
    as="button"
    icon={true}
    onClick={[Function]}
  >
    <Icon
      as="i"
      name="arrow left"
    />
  </Button>
  <Container>
    <Grid
      padded={true}
    >
      <Header
        as="h2"
        icon={true}
        textAlign="center"
      >
        <Icon
          as="i"
          name="github"
        />
        <HeaderContent>
          Some Repo
        </HeaderContent>
      </Header>
      <GridRow
        columns={3}
      >
        <GridColumn>
          <Connect(PrTable) />
        </GridColumn>
        <GridColumn>
          <Connect(Issues) />
        </GridColumn>
        <GridColumn>
          <Connect(Milestones) />
        </GridColumn>
      </GridRow>
    </Grid>
  </Container>
</div>
`;
